src/cdomains/addressDomain_intf.ml

```diff
@@ -0,0 +1,117 @@
+module type AddressDomain =
+sig
+  module AddressBase (Mval: Printable.S):
+  sig
+    type t =
+      | Addr of Mval.t (** Pointer to offset of a variable. *)
+      | NullPtr (** NULL pointer. *)
+      | UnknownPtr (** Unknown pointer. Could point to globals, heap and escaped variables. *)
+      | StrPtr of string option (** String literal pointer. [StrPtr None] abstracts any string pointer *)
+    include Printable.S with type t := t (** @closed *)
+    include MapDomain.Groupable with type t := t (** @closed *)
+
+    val of_string: string -> t
+    val to_string: t -> string option
+    val to_c_string: t -> string option
+    val to_n_c_string: int -> t -> string option
+    val to_string_length: t -> int option
+  end
+
+  module AddressPrintable (Mval: Mval.Printable):
+  sig
+    include module type of AddressBase (Mval)
+    include MapDomain.Groupable with type t := t and type group = Basetype.Variables.group (** @closed *)
+
+    val is_definite: t -> bool
+    val add_offset: t -> Mval.idx Offset.t -> t
+
+    val of_var: GoblintCil.varinfo -> t
+    (** Creates an address from variable. *)
+
+    val of_mval: Mval.t -> t
+    (** Creates an address from a variable and offset. *)
+
+    val to_var: t -> GoblintCil.varinfo option
+    (** Strips the varinfo out of the address representation. *)
+
+    val to_var_may: t -> GoblintCil.varinfo option
+    val to_var_must: t -> GoblintCil.varinfo option
+    (** Strips the varinfo out of the address representation. *)
+
+    val to_mval: t -> Mval.t option
+    (** Get the offset *)
+
+    val to_exp: t -> GoblintCil.exp
+
+    val type_of: t -> GoblintCil.typ
+    (** Finds the type of the address location. *)
+  end
+
+  (** Lvalue lattice.
```

Comment:
Is this still an `Lvalue lattice`?

Reply:
I just copied the old documentation. I've now updated it.